### PR TITLE
perf: reduce memory usage

### PR DESCRIPTION
**Pull request by evilebottnawi**
We create source map on each `file` and it is increase memory usage very very very lagre, but using `sourceMap` only for error. Let's create `sourceMap` only when erros/warnings were throws.

### CODE REVIEW

**Comment by evilebottnawi** [comment]
@TheLarkInn no issue, it is own metrics in our private projects. We speedup plugins (because now we don't spend time on parse source maps everytime, because we need in they only when error occurs), also don't use memory for this.

File: src/index.js [hunk 1]
```diff
@@ -66,21 +73,29 @@ class UglifyJsPlugin {
...
-      const original = sourceMap.originalPositionFor({
+      const original = sourceMap && sourceMap.originalPositionFor({
         line,
         column,
       });
 
       if (original && original.source && original.source !== file && warningsFilter(original.source)) {
         accWarnings.push(`${warning.replace(warningRegex, '')}[${requestShortener.shorten(original.source)}:${original.line},${original.column}]`);
+      } else {
+        accWarnings.push(`${warning.replace(warningRegex, '')}`);
```
Comment (evilebottnawi) [inline src/index.js]: Fix, when we don't have source map warnings not emitting :confused:

File: test/__snapshots__/all-options.test.js.snap [hunk 2]
```diff
@@ -57,7 +57,10 @@ exports[`when applied with all options matches snapshot: manifest.d6857f782c13a9
 
 exports[`when applied with all options matches snapshot: warnings 1`] = `
 Array [
-  "Error: main.0c220ec66316af2c1b24.js from UglifyJs
-Dropping unused variable a [./test/fixtures/entry.js:4,0]",
+  Array [
+    "Error: main.27965655b499b10158f3.js from UglifyJs",
+    "Dropping unused variable a [./test/fixtures/entry.js:4,0]",
+    "Dropping unused variable c [./test/fixtures/entry.js:5,0]",
```
Comment (evilebottnawi) [inline test/__snapshots__/all-options.test.js.snap]: Fix error helper in tests. Now we can see all warnings in tests.

**Comment by evilebottnawi** [comment]
@d3viant0ne need patch release :+1:

**Comment by BoldBigflank** [comment]
I'm trying out 1.2.5 and still seeing a hang at 92% (I thought it was working but it was just failing at a missing dependency at first)

**Pull request merged**